mDNSWindows/mDNSWin32.c

```diff
@@ -1054,7 +1054,7 @@ mDNSPlatformTCPConnect
 
 	// Setup connection data object
 
-	sock->userCallback	= inCallback;
+	sock->userCallback	= (void*) inCallback;
```

Comment:
Why isn't TCPUserCallback of type TCPConnectionCallback?

Reply:
It has different arguments.

`warning: incompatible function pointer types assigning to 'TCPUserCallback' (aka 'void (*)()') from 'TCPConnectionCallback' (aka 'void (*)(struct TCPSocket_struct *, void *, unsigned char, int)') [-Wincompatible-function-pointer-types]`

Reply:
Right, but isn't that bad? Shouldn't it have the same arguments?

Reply:
Isn't it a C-style way of passing user callbacks? I'm pretty sure this was exactly the way it was done in other C open source project I worked on.

So [TCPUserCallback](https://github.com/Abhayakara/mdnsresponder/blob/50902620f335639ffc86d0bc41c101e684390f70/mDNSWindows/mDNSWin32.h#L35) is just a function pointer equivalent to `void*` and [TCPConnectionCallback](https://github.com/Abhayakara/mdnsresponder/blob/50902620f335639ffc86d0bc41c101e684390f70/mDNSCore/mDNSEmbeddedAPI.h#L3050) is a concrete callback implementation.

Reply:
Having said that, I should have casted it to that pointer type, not to `void*`. Pushed an updated change that matches what's done in the other place (line 1223, function `TCPSocketNotification`).

Reply:
Why not just delete this:

mDNSWin32.h:typedef void ( *TCPUserCallback )();

and change this:

mDNSWin32.h:	TCPUserCallback				userCallback;
 
to this:

mDNSWin32.h:	TCPConnectionCallback				userCallback;

Reply:
This is exactly what I tried first, but then I got build error:

```
external/mdnsresponder/mDNSWindows/SystemService/Service.c:2023:68: error: too few arguments to function call, expected 4, have 2
                tcpSock->userCallback( ( int ) tcpSock->fd, tcpSock->userContext );
                ~~~~~~~~~~~~~~~~~~~~~                                            ^
external/mdnsresponder/mDNSWindows/SystemService/Service.c:2050:23: warning: incompatible function pointer types assigning to 'TCPConnectionCallback' (aka 'void (*)(struct TCPSocket_struct *, void *, unsigned char, int)') from 'udsEventCallback' (aka 'void (*)(int, void *)') [-Wincompatible-function-pointer-types]
                sock->userCallback              = callback;
                                                ^ ~~~~~~~~
```

Because of what I described in the other comment thread (why there is two btw?).